packages/syft/src/syft/core/node/new/sqlite_document_store.py

```diff
@@ -201,16 +201,11 @@ def _delete_all(self) -> None:
         self._execute(select_sql)
 
     def _len(self) -> int:
-        select_sql = f"select uid from {self.table_name}"  # nosec
-        uids = self._execute(select_sql)
+        select_sql = f"select * from {self.table_name}"  # nosec
```

Comment:
The code on the left only iterates through the cursor which might be quicker than getting all the data if fetchall does that.